src/coordinator/roles.spec.js

```diff
@@ -49,8 +49,164 @@ describe('Coordinator/Roles', function() {
+        };
+
+        newRolePayloadToServer = {
+          name: newRolePayload.name,
+          description: newRolePayload.description
+        };
+
+        request = {
+          ...baseRequest,
+          post: this.sandbox.stub().resolves(expectedRoleFromServer)
+        };
+        toCamelCase = this.sandbox
+          .stub(objectUtils, 'toCamelCase')
+          .callsFake(() => expectedRole);
+
+        toSnakeCase = this.sandbox
+          .stub(objectUtils, 'toSnakeCase')
+          .callsFake(() => newRolePayloadToServer);
+
+        roles = new Roles(baseSdk, request, expectedHost);
+        promise = roles.create(organization.id, newRolePayload);
+      });
+
+      it('formats the role payload', function() {
+        return promise.then(() => {
+          expect(toSnakeCase).to.be.calledWith(newRolePayload);
+        });
+      });
+
+      it('posts the role to the server', function() {
+        expect(request.post).to.be.calledOnce;
+        expect(request.post).to.be.calledWith(
+          `${expectedHost}/organizations/${organization.id}/roles`,
+          newRolePayloadToServer
+        );
+      });
+
+      it('returns a fulfilled promise', function() {
+        return expect(promise).to.be.fulfilled.and.to.eventually.deep.equal(
+          expectedRole
+        );
+      });
+
+      it('formats the role response', function() {
+        return promise.then(() => {
+          expect(toCamelCase).to.be.calledWith(expectedRoleFromServer);
+        });
+      });
+    });
+
+    context('when the organizationId is not provided', function() {
+      it('returns a rejected promise', function() {
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const promise = roles.create();
+
+        return expect(promise).to.be.rejectedWith(
+          'An organizationId is required for creating roles for an organization.'
+        );
+      });
+    });
+
+    context('when the role does not have required properties', function() {
+      it('without name it returns a rejected promise', function() {
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const organization = fixture.build('contxtOrganization');
+        const promise = roles.create(organization.id, {
+          description: 'winning'
+        });
+
+        return expect(promise).to.be.rejectedWith(
+          `A name is required to create a new role.`
+        );
+      });
+      it('without description it returns a rejected promise', function() {
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const organization = fixture.build('contxtOrganization');
+        const promise = roles.create(organization.id, { name: 'winning' });
+
+        return expect(promise).to.be.rejectedWith(
+          `A description is required to create a new role.`
+        );
+      });
+    });
+  });
+
+  describe('delete', function() {
+    context('when the organizationId is provided', function() {
```

Comment:
This should probably be similar and say something like `when the required information is provided`. Right now it sounds like just when `organizationId` is provided (which should error since `roleId` wouldn't be provided)

Reply:
done